services/core-api/app/api/users/minespace/resources/minespace_user.py

```diff
@@ -20,14 +21,26 @@ class MinespaceUserListResource(Resource, UserMixin):
     parser.add_argument('mine_guids', type=list, location='json', required=True)
 
     @api.doc(params={
-        'email_or_username': 'find by email, this will return a list with at most one element'
+        'email_or_username': 'find by email, this will return a list with at most one element',
+        'mine_guid': 'find by mine guid, this will return all users with access to the specified mine'
     })
     @api.marshal_with(MINESPACE_USER_MODEL, envelope='records')
-    @requires_role_mine_admin
+    @requires_any_of([VIEW_ALL, MINESPACE_PROPONENT])
     def get(self):
+        mine_guid = request.args.get('mine_guid')
+        is_admin = getJwtManager().contains_role([MINE_ADMIN])
+
+        if not is_admin and mine_guid is None:
+            raise BadRequest("mine_guid is a required argument")
+        
         if request.args.get('email_or_username'):
             ms_users = [MinespaceUser.find_by_email(request.args.get('email_or_username'))]
-        else:
+        elif mine_guid:
+            mine = Mine.find_by_mine_guid(mine_guid)
```

Comment:
It might be good to add a check to make sure the user fetching the list is included in the returned list (if they aren't an admin)